### ISSUE TITLE
Calendar not password protected

### ISSUE THREAD

**Issue by PBue** [nextcloud/calendar#694]
Hi, this is my first report so I hope I did not forget anything. Thank you!
### Steps to reproduce
1. Create new calendar named "Test"
2. Do not click share or anything else
3. Configure Thunderbird to subscribe with CalDAV and URL: https://domain.tld/nextcloud/remote.php/dav/calendars/yourusername/test/

### Expected behaviour
There should be a password protection for calendar "Test" otherwise anyone could just guess my calendars name and subscribe to it.

### Actual behaviour
Thunderbird shows all entries in "Test" without asking for username and password.

### Server configuration
**Operating system:** Linux 4.13.8-3-pve #1 SMP PVE 4.13.8-30 (Tue, 5 Dec 2017 13:06:48 +0100) x86_64

**Webserver:** Apache/2.4.25 (Debian) (apache2handler)

**Database:** mysql 10.1.26

**PHP version:** 7.0.19-1
Modules loaded: Core, date, libxml, openssl, pcre, zlib, filter, hash, Reflection, SPL, session, standard, apache2handler, mysqlnd, PDO, xml, apcu, apc, calendar, ctype, curl, dom, mbstring, fileinfo, ftp, gd, gettext, iconv, imagick, intl, json, exif, mcrypt, mysqli, pdo_mysql, Phar, posix, readline, shmop, SimpleXML, sockets, sysvmsg, sysvsem, sysvshm, tokenizer, wddx, xmlreader, xmlwriter, xsl, zip, Zend OPcache

**Nextcloud version:** 12.0.4 - 12.0.4.3

**Updated from an older Nextcloud/ownCloud or fresh install:** 

**Where did you install Nextcloud from:** unknown

<details><summary>Signing status</summary>

Array
</details>

<details><summary>List of activated apps</summary>

```
Enabled:
 - activity: 2.5.2
 - bruteforcesettings: 1.0.3
 - calendar: 1.5.7
 - comments: 1.2.0
 - contacts: 2.0.1
 - dav: 1.3.0
 - encryption: 1.6.0
 - federatedfilesharing: 1.2.0
 - federation: 1.2.0
 - files: 1.7.2
 - files_pdfviewer: 1.1.1
 - files_reader: 1.0.4
 - files_sharing: 1.4.0
 - files_texteditor: 2.4.1
 - files_trashbin: 1.2.0
 - files_versions: 1.5.0
 - files_videoplayer: 1.1.0
 - firstrunwizard: 2.1
 - gallery: 17.0.0
 - issuetemplate: 0.3.0
 - keeweb: 0.4.0
 - logreader: 2.0.0
 - lookup_server_connector: 1.0.0
 - nextcloud_announcements: 1.1
 - notifications: 2.0.0
 - oauth2: 1.0.5
 - password_policy: 1.2.2
 - provisioning_api: 1.2.0
 - ransomware_protection: 1.0.5
 - serverinfo: 1.2.0
 - sharebymail: 1.2.0
 - survey_client: 1.0.0
 - systemtags: 1.2.0
 - tasks: 0.9.5
 - theming: 1.3.0
 - twofactor_backupcodes: 1.1.1
 - updatenotification: 1.2.0
 - workflowengine: 1.2.0
Disabled:
 - admin_audit
 - files_external
 - user_external
 - user_ldap

```
</details>

<details><summary>Configuration (config/config.php)</summary>

```
{
    "instanceid": "***REMOVED SENSITIVE VALUE***",
    "passwordsalt": "***REMOVED SENSITIVE VALUE***",
    "secret": "***REMOVED SENSITIVE VALUE***",
    "trusted_domains": [
        "***REMOVED SENSITIVE VALUE***",
        "***REMOVED SENSITIVE VALUE***"
    ],
    "datadirectory": "\/nextcloud\/data",
    "overwrite.cli.url": "***REMOVED SENSITIVE VALUE***",
    "dbtype": "mysql",
    "version": "12.0.4.3",
    "dbname": "nextcloud",
    "dbhost": "localhost",
    "dbport": "",
    "dbtableprefix": "oc_",
    "dbuser": "***REMOVED SENSITIVE VALUE***",
    "dbpassword": "***REMOVED SENSITIVE VALUE***",
    "installed": true,
    "memcache.local": "\\OC\\Memcache\\APCu",
    "maintenance": false
}
```
</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...

**Are you using encryption:** yes

## Client configuration

**Browser:** Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/61.0.3163.100 Safari/537.36 OPR/48.0.2685.52

**Comment by georgehrke** [comment]
> Hi, this is my first report so I hope I did not forget anything. Thank you!

Hey, if you are reporting security issues, please never ever post them into the public bug tracker.
See https://nextcloud.com/security/ for more info.

Second of all, as the comment in the issue template states, please report CalDAV server issues like this in the server repo, not the calendar repo.


Nevertheless, I'm not able to reproduce this issue against my private Nextcloud 12.0.4 instance. It always asks me for authentication: 

![bcc44ca6-c1d1-4406-b63c-4eafca6fc0b6](https://user-images.githubusercontent.com/1250540/34641724-c9b0236e-f308-11e7-9489-db9288b54dae.jpg)

**Comment by georgehrke** [comment]
Are there any other CalDAV calendars set up in Thunderbird?

**Comment by PBue** [comment]
> Hey, if you are reporting security issues, please never ever post them into the public bug tracker.
See https://nextcloud.com/security/ for more info.

> Second of all, as the comment in the issue template states, please report CalDAV server issues like this in the server repo, not the calendar repo.
Sorry for this!

> Are there any other CalDAV calendars set up in Thunderbird?
No, there is also a clear password manager and I restarted Thunderbird a few times.

I reinstalled Thunderbird completly and it works now.
I think I found the mistake in my own configuration, because I had my calendar set up in Thunderbird before (and removed it) a few weeks ago. Guess Thunderbird cached the password elsewhere so I could not find it in the password manager.

I am so sorry for the inconvenience I caused!
